app/src/behaviors/behavior_capslock.c

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2021 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_capslock
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+#include <zmk/behavior.h>
+
+#include <zmk/behavior_queue.h>
+#include <zmk/endpoints.h>
+#include <zmk/event_manager.h>
+#include <zmk/events/keycode_state_changed.h>
+#include <zmk/hid.h>
+#include <zmk/keys.h>
+#include <zmk/keymap.h>
+#include <zmk/led_indicators.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
+
+struct capslock_key_item {
+    uint16_t page;
+    uint32_t id;
+    uint8_t implicit_modifiers;
+};
+
+struct behavior_capslock_config {
+    uint8_t index;
+    struct zmk_behavior_binding capslock_binding;
+    bool enable_on_press;
+    bool disable_on_release;
+    bool disable_on_next_release;
+    uint8_t disable_on_keys_count;
+    struct capslock_key_item disable_on_keys[];
+};
+
+struct behavior_capslock_data {
+    uint32_t position;
+    bool active;
+    bool just_activated;
+};
+
+static void toggle_capslock(const struct device *dev) {
+    const struct behavior_capslock_config *config = dev->config;
+    const struct behavior_capslock_data *data = dev->data;
+
+    zmk_behavior_queue_add(data->position, config->capslock_binding, true, 0);
```

Comment:
```suggestion
    zmk_behavior_queue_add(data->position, config->capslock_binding, true, 100);
```

@snoyer @petejohanson I dug into the MacOS caps lock delay issue. Indeed, it's a "feature" of MacOS that you  must hold down caps lock briefly to activate it. I started with this wait number at 500ms to confirm that adding such a delay to the capsword code would work, then backed it down until I found the minimum that would reliably work. (The actual threshold seems to be around 75ms.) In my testing this delay does not seem to be noticeable at all.

Reply:
Thanks, @rschenk, I just pushed a similar change before I saw this comment.